src/main/java/edu/ksu/canvas/requestOptions/BaseOptions.java

```diff
@@ -30,4 +31,8 @@ protected void addSingleItem(String key, String value) {
     protected void addNumberList(String key, List<? extends Number> list) {
         optionsMap.put(key, list.stream().map(i->i.toString()).collect(Collectors.toList()));
     }
+
+    protected void addStringList(String key, List<String> list) {
+        optionsMap.put(key, new ArrayList<String>(list));
```

Comment:
Any particular reason for allocating a new ArrayList object instead of just passing the list straight in?

Reply:
Ok, fair enough. Strictly speaking this method isn't needed as some other options classes just directly add string lists to the `optionsMap` but it does make things more consistent so I may go refactor the other classes to call this method after I merge it.

Reply:
Frankly I'm not a huge fan of defensive copying, but when the 3 existing methods exhibited a behaviour I figured it was better to keep it. Plus as long as all these method are doing it, then there's no need to do so in the Options classes.